### ISSUE TITLE
autofix collision: cache-poisoning-poisonable-step deletes the 'with: ref:' that untrusted-checkout-toctou/critical just fixed

### ISSUE THREAD

**Issue by on-keyday** [ultra-supara/sisakulint#533]
## Summary

`untrusted-checkout-toctou/critical` and `cache-poisoning-poisonable-step` both autofix the same `actions/checkout` step, with contradictory strategies:

- TOCTOU: replace the mutable ref (`github.event.pull_request.head.ref` / `github.head_ref`) with `github.event.pull_request.head.sha` — keep the `ref:` input, make it immutable.
- cache-poisoning: delete the `with: ref:` block entirely and rely on the default ref.

Because the fixers run sequentially on the same step AST, whichever runs later silently wins. In the default run, cache-poisoning's deletion erases the `ref:` input that TOCTOU just rewrote — the head.sha replacement never appears in the output, and no warning is emitted. Affected version: `main` @ `dfeb590` (re-verified 2026-07-07).

## Reproduce

```bash
go build -o ./sisakulint ./cmd/sisakulint
./sisakulint -fix dry-run script/actions/untrusted-checkout-toctou-critical.yaml
```

The fixture has two checkout steps with `ref: ${{ github.event.pull_request.head.ref }}` / `ref: ${{ github.head_ref }}`. In the fixed output both become a bare `uses: actions/checkout@v4` — no `with:`, no `ref:`, no head.sha.

Note the resulting workflow is not wrong for the TOCTOU concern (checking out the default ref also removes the mutable-ref race), but the two rules disagree about the intended shape of the fix, and TOCTOU's documented Before/After can never be observed when both rules are enabled.

## Suggested direction

Same root class as the epic's pattern 1 (fixer composition on a shared AST), but the collision is on the sink node itself rather than env-var naming: the fixers need either an agreed canonical strategy for steps where both fire (probably: deleting the input wins, and TOCTOU should then skip and not advertise its own rewrite), or an ordering/awareness mechanism so the surviving fix is explicit rather than accidental.

Related to #502 (same composition pattern, outside the injection family).
